### PR TITLE
ABI dlopen example

### PR DESCRIPTION
**Pull request by jjhursey**

### CODE REVIEW

**Comment by jjhursey** [comment]
Depends on https://github.com/pmix/pmix-abi/pull/9 (which is also shown in this PR).

**Comment by jjhursey** [comment]
Rebased now that PR #9 has been merged.

File: example/abi-query-dlopen.c [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    PMIX_ARGV_APPEND(rc, query[0].keys, PMIX_QUERY_ABI_VERSION);
+
+    //rc = PMIx_Query_info(query, nqueries, &info, &ninfo);
+    rc = query_fn(query, nqueries, &info, &ninfo);
+    if (PMIX_SUCCESS != rc ) {
+        fprintf(stderr, "Error: PMIx_Query_info failed: %d\n", rc);
+        return rc;
+    }
+
+    printf("--> Query returned (ninfo %d)\n", ninfo);
```
Comment (raffenet) [inline example/abi-query-dlopen.c:84]:
```suggestion
    printf("--> Query returned (ninfo %lu)\n", ninfo);
```
squash a `-Wformat` warning

**Review by raffenet — APPROVED** [review]
LGTM. One minor suggestion to fix a warning.

File: example/abi-query-dlopen.c (hunk 1 above)
Comment (jjhursey) [reply example/abi-query-dlopen.c]: fixed. thanks!

File: example/abi-query-dlopen.c [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        printf("--> KEY: %s\n", info[i].key);
+        if (PMIX_CHECK_KEY(&info[i], PMIX_QUERY_ABI_VERSION)) {
+            printf("----> ABI: String: %s\n",
+                   info[i].value.data.string);
+        }
+    }
+
+    PMIX_INFO_FREE(info, ninfo);
+    PMIX_QUERY_FREE(query, nqueries);
+
```
Comment (raffenet) [inline example/abi-query-dlopen.c:95]: nit: could add
```
dlclose(lib_handle);
```
before returning, but either way it's fine.

Comment (jjhursey) [reply example/abi-query-dlopen.c:95]: Fixed. Thanks for catching that.

**Comment by jjhursey** [comment]
I just force pushed to this branch to update the example to reflect the stable/provisional split in the current proposal (https://github.com/pmix/pmix-standard/pull/397). The OpenPMIx implementation is here: https://github.com/openpmix/openpmix/pull/2617

**Comment by jjhursey** [comment]
Per IAWG meeting July 11, 2022:
 * We should add a lookup table in the base `README.md` to mark the version pairing on the PMIx Standard with the PMIx ABI.
 * Once PMIx Standard of v5.0 is released then we will mark the stable and provisional ABI to v1.0.
 * (Josh) Will add to this PR the README.md table

**Comment by jjhursey** [comment]
I just pushed the first draft of the version table to this PR. It looks like:
image

**Comment by jjhursey** [comment]
All PMIx Governance and PMIx Standard items have been merged for the ABI. We can merge this now and update the `README.md` with any notes necessary when we release the PMIx Standard v5.0.

**Pull request merged**